### ISSUE TITLE
Assessing Direct Monitoring Techniques to Analyze Failures of Critical Industrial Systems

### ISSUE THREAD

**Issue by vivekaxl** [opensciences/opensciences.github.io#144]
Paper: http://ieeexplore.ieee.org/search/searchresult.jsp?newsearch=true&queryText=Assessing+Direct+Monitoring+Techniques+to+Analyze+Failures+of+Critical+Industrial+Systems

Dataset: http://www.mobilab.unina.it/Datasets.htm

**Comment by WeiFoo** [comment]
author: Marcello Cinque, Domenico Cotroneo, Raffaele Della Corte, Antonio Pecchia
email:{macinque, cotroneo, raffaele.dellacorte2, antonio.pecchia}@unina.it

**Comment by reesjones** [comment]
Needs new context notes. See [this example](https://github.com/opensciences/opensciences.github.io/issues/160) or [this example](http://openscience.us/repo/issues/bugsal.html) as a reference.

Context notes are directly inserted into the actual repository (like [this](http://openscience.us/repo/issues/bugsal.html)), so good context notes are necessary for putting the datasets into the repo in a timely fashion.

**Comment by WeiFoo** [comment]
**Categorization**

Direct Monitoring Dataset


**Author list for the paper**

author: Marcello Cinque, Domenico Cotroneo, Raffaele Della Corte, Antonio Pecchia
email:{macinque@unina.it , cotroneo@unina.it , raffaele.dellacorte2@unina.it , antonio.pecchia@unina.it }

**General overview of the data**

Direct Monitoring Dataset is a collection of data obtained during an experimental analysis of different direct monitoring techniques.
The analysis based on a fault-injection campaign on a real-world air traffic control (ATC) middleware system conducted by SAFE tool. The transport layer of the considered middleware implements three direct monitoring techniques, i.e., event logs, assertions, and rule-based loging approach supported by LogBus. We injected 12,733 software faults, which types extend the Orthogonal Defect Classification (ODC) for practical injection purposes, into the middleware transport layer source code. For each injected faults, we collected the data generated by considered techniques at middleware level, but we also collected the event logs produced by the operating system (Red Hat 5 El) and the legacy ATC applications running on the top of the target middleware. In summary, collected data include:

data generated by middleware:
logs (MwLog)
assertions (MwAss);
rule-based logs (MwR-BLog);
operating system logs (OsLog);
application logs (AppLog).
The dataset contains a folder for each technique mentioned above. Each folder contains 12,733 files, a file for each experiments (i.e., faults injected in the middleware). Each file cotains the data generated by technique when we inject that fault in the middleware transport layer source code. The name of file indicates the source code file where the fault is injected and the type of fault.

**Link to the paper associated with the dataset**

http://ieeexplore.ieee.org/xpl/articleDetails.jsp?tp=&arnumber=6982628&queryText%3DAssessing+Direct+Monitoring+Techniques+to+Analyze+Failures+of+Critical+Industrial+Systems

**Paper abstract**

The analysis of monitoring data is extremely valuable for critical computer systems. It allows to gain insights into the failure behavior of a given system under real workload conditions, which is crucial to assure service continuity and downtime reduction.
This paper proposes an experimental evaluation of different direct monitoring techniques, namely event logs, assertions, and source code instrumentation, that are widely used in the context of critical industrial systems. We inject 12,733 software faults in a real-world air traffic control (ATC) middleware system with the aim of analyzing the ability of mentioned techniques to produce information in case of failures. Experimental results indicate that each technique is able to cover a limited number of failure manifestations. Moreover, we observe that the quality of collected data to support failure diagnosis tasks strongly varies across the techniques considered in this study.

**BibTeX reference for the paper**
@INPROCEEDINGS{6982628, 
author={Cinque, M. and Cotroneo, D. and Della Corte, R. and Pecchia, A.}, 
booktitle={Software Reliability Engineering (ISSRE), 2014 IEEE 25th International Symposium on}, 
title={Assessing Direct Monitoring Techniques to Analyze Failures of Critical Industrial Systems}, 
year={2014}, 
month={Nov}, 
pages={212-222}, }

**Link to the datasets**
http://www.mobilab.unina.it/Datasets.htm


Is this dataset part of a larger series or collection?
no
